### PR TITLE
Attempt to fix code making pods OOM

### PR DESCRIPTION
**Pull request by ryantk**
1. Reapplied the PR that was revered (#2528)
2. I found some code which in effect was calling prisoner search API multiple times
3. Rewrote a method which was "caching" potentially huge objects to be a simple find

### CODE REVIEW

File: app/controllers/prisoners_controller.rb [hunk 1]
```diff
@@ -114,22 +121,6 @@ def load_summary(summary_type)
     @offenders = sort_and_paginate(items, default_sort: :last_name)
   end
 
-  def get_slice_for_page(offender_list, user_id)
-    offenders = []
-    user_allocations = []
-
-    offender_list.map do |offender|
-      allocation = @prison.allocation_for(offender)
```
Comment (ryantk) [inline app/controllers/prisoners_controller.rb]: `@prison.allocation_for(offender)` Loads a `Prison::AllocationSummary` which is populated with `Prison#policy_offenders` - 1 call to prisoner search here

File: app/controllers/prisoners_controller.rb [hunk 2]
```diff
@@ -21,11 +23,16 @@ def unallocated
 
   def search
     @q = search_term
-    offenders = SearchService.search_for_offenders(@q, @prison.policy_offenders)
```
Comment (ryantk) [inline app/controllers/prisoners_controller.rb]: 1 call to prisoner search here

File: app/controllers/prisoners_controller.rb [hunk 3]
```diff
@@ -21,11 +23,16 @@ def unallocated
...
-    @offenders, @user_allocations = get_slice_for_page(offenders, @current_user.staff_id)
-    MetricsService.instance.increment_search_count
 
-    render @current_user.allocations.empty? ? 'search' : 'search_global'
```
Comment (ryantk) [inline app/controllers/prisoners_controller.rb:28]: `@current_user.allocations.empty?` Will call prisoner search again with its own `Prison::AllocationSummary`

**Review by zheileman — APPROVED** [review]
LGTM! 😅

**Pull request merged**